src/Hint/Eval.hs

```diff
@@ -30,7 +31,7 @@ infer = undefined
 
 -- | Evaluates an expression, given a witness for its monomorphic type.
 interpret :: (MonadInterpreter m, Typeable a) => String -> a -> m a
-interpret expr wit = unsafeInterpret expr (show $ typeOf wit)
+interpret expr wit = unsafeInterpret expr (show $ Typeable.typeOf wit)
```

Comment:
Even though there is only one `typeOf` in this module, I think it's better to keep it qualified in order to make it clear that unlike every other occurrence of `typeOf` in this package, this is `Data.Typeable.typeOf`, not `Language.Haskell.Interpreter.typeOf`.